src/ascii/setting.rs

```diff
@@ -0,0 +1,293 @@
+//! Types and traits for accessing device or axis settings
+
+include!("setting/mod.inc");
```

Comment:
crate takes a _lot_ longer to process now.
- `cargo check`: ~10 s
- `cargo doc --no-deps`: ~3 s
- `cargo clippy`: ~16 s
-  `cargo build`: ~16 s

Reply:
Output of `cargo clean && cargo build --timings` (see cargo --timings [docs](https://doc.rust-lang.org/nightly/cargo/reference/timings.html)). 
```
Targets:	zproto 0.3.5-pre (lib)
Profile:	dev
Fresh units:	0
Dirty units:	20
Total units:	20
Max concurrency:	8 (jobs=8 ncpu=8)
Build start:	2024-01-13T20:28:20Z
Total time:	16.1s
rustc:	rustc 1.75.0 (82e1608df 2023-12-21)
Host: x86_64-unknown-linux-gnu
Target: x86_64-unknown-linux-gnu
```

![build-tree](https://github.com/stphnt/zproto/assets/56858839/8e0731b5-885f-4fd9-9532-1e790a06e1b3)
![units-available](https://github.com/stphnt/zproto/assets/56858839/d035b982-8fcc-43af-913f-bc2f57123e2c)

Reply:
For comparison, the previous output is:

```
Targets:	zproto 0.3.5-pre (lib)
Profile:	dev
Fresh units:	0
Dirty units:	20
Total units:	20
Max concurrency:	8 (jobs=8 ncpu=8)
Build start:	2024-01-13T23:09:01Z
Total time:	4.2s
rustc:	rustc 1.75.0 (82e1608df 2023-12-21)
Host: x86_64-unknown-linux-gnu
Target: x86_64-unknown-linux-gnu
```
![build-tree-prev](https://github.com/stphnt/zproto/assets/56858839/aa67c93f-168d-410e-8434-68fe17aab3ea)
![units-available-prev](https://github.com/stphnt/zproto/assets/56858839/341bc6c5-b0c9-48c8-b323-a4f80b046e3d)

Codegen (lavender) is ~22% of zproto's build time.

Reply:
After adding cargo features for each firmware version, build times have been reduced to 3.5-4s. Much better than it was before.